### DISCUSSION TITLE
How many basic 'Contribution' are there in projects

### DISCUSSION THREAD

**Discussion by YeomanYe** [eclipse-theia/theia#9649] [Q&A]
I want to learn the architecture of Theia, so I want to know how many basic `Contribution` are in the project of `Theia`, but I don't see it in the document. Is there documentation for this content? Or is there a way to get the answer

**Comment by vince-fugnitto** [comment]
@YeomanYe thank you for the discussion, there are many different types of contribution to the framework, defined either by the `core` of the framework or extensions. When you reference `basic`, what exactly are you referring to?

You can take a look at references of `ContributionProvider` in the codebase, but there are many instances.

For the most simple or common there is:
- [MenuContribution](https://eclipse-theia.github.io/theia/docs/next/interfaces/core.menucontribution-1.html)
- [CommandContribution](https://eclipse-theia.github.io/theia/docs/next/interfaces/core.commandcontribution-1.html)
- [KeybindingContribution](https://eclipse-theia.github.io/theia/docs/next/interfaces/core.keybindingcontribution-1.html)
- [AbstractViewContribution](https://eclipse-theia.github.io/theia/docs/next/classes/core.abstractviewcontribution-2.html)

You can search for `contribution` in the [api documentation](https://eclipse-theia.github.io/theia/docs/next/modules/core.html) of `@theia/core` which is the parent most extension which all other extensions depend on.
